WordPress/src/main/java/org/wordpress/android/ui/reader/discover/interests/ReaderInterestsViewModel.kt

```diff
@@ -1,6 +1,60 @@
 package org.wordpress.android.ui.reader.discover.interests
 
-import androidx.lifecycle.ViewModel
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.launch
+import org.wordpress.android.models.ReaderTagList
+import org.wordpress.android.modules.UI_THREAD
+import org.wordpress.android.ui.reader.repository.ReaderTagRepository
+import org.wordpress.android.viewmodel.ScopedViewModel
 import javax.inject.Inject
+import javax.inject.Named
 
-class ReaderInterestsViewModel @Inject constructor() : ViewModel()
+class ReaderInterestsViewModel @Inject constructor(
+    @Named(UI_THREAD) private val mainDispatcher: CoroutineDispatcher,
+    private val readerTagRepository: ReaderTagRepository
+) : ScopedViewModel(mainDispatcher) {
+    var initialized: Boolean = false
+
+    private val _uiState = MutableLiveData<UiState>()
+    val uiState: LiveData<UiState> = _uiState
+
+    fun start() {
+        if (initialized) return
+        loadInterests()
+    }
+
+    private fun loadInterests() {
+        launch {
+            val tagList = readerTagRepository.getInterests()
+            if (tagList.isNotEmpty()) {
+                updateUiState(UiState(transformToInterestsUiState(tagList), tagList))
+                if (!initialized) {
```

Comment:
I haven't seen this usage of "initialized" anywhere else in the app. I'm not sure it'll still make sense when we introduce loading/error handling. But we'll see :).

Reply:
I found `initialized` used in the [ReaderViewModel's start method](https://github.com/wordpress-mobile/WordPress-Android/blob/develop/WordPress/src/main/java/org/wordpress/android/ui/reader/viewmodels/ReaderViewModel.kt#L62).

>  I'm not sure it'll still make sense when we introduce loading/error handling. But we'll see :).

👍

Reply:
:D ohh, 🐟 memory. 

👍